pygameMenu/menu.py

```diff
@@ -1,1614 +1,1581 @@
+        for widget in self._actual._widgets:
+            widget.draw(self._actual._widgets_surface)
+            if self._actual._selection_highlight and widget.selected:  # If selected draw a rectangle
+                widget.draw_selected_rect(self._actual._widgets_surface,
+                                          self._actual._selection_color,
+                                          self._actual._selection_highlight_margin_x,
+                                          self._actual._selection_highlight_margin_y,
+                                          self._actual._selection_border_width)
+
+        self._actual._scroll.draw(surface)
+        self._actual._menubar.draw(surface)
+
+    def enable(self):
+        """
+        Enables Menu (can check events and draw).
+
+        :return: None
+        """
+        if not self._top._enabled:
+            self._top._enabled = True
+            self._top._closelocked = True
+
+    @staticmethod
+    def _exit():
+        """
+        Internal exit function.
+
+        :return: None
+        """
+        _pygame.quit()
+        sys.exit()
+
+    def is_enabled(self):
+        """
+        Returns True if menu is enabled else False is returned.
+
+        :return: True if the menu is enabled
+        :rtype: bool
+        """
+        return self._top._enabled
+
+    def _left(self):
+        """
+        Left event (column support).
+        """
+        if self._index >= self._rows:
+            self._select(self._index - self._rows)
+        else:
+            self._select(0)
+
+    def _right(self):
+        """
+        Right event (column support).
+        """
+        if self._index + self._rows < len(self._widgets):
+            self._select(self._index + self._rows)
+        else:
+            self._select(len(self._widgets) - 1)
+
+    def _handle_joy_event(self):
+        """
+        Handle joy events.
+        """
+        if self._joy_event & _JOY_EVENT_UP:
+            self._select(self._index - 1)
+        if self._joy_event & _JOY_EVENT_DOWN:
+            self._select(self._index + 1)
+        if self._joy_event & _JOY_EVENT_LEFT:
+            self._left()
+        if self._joy_event & _JOY_EVENT_RIGHT:
+            self._right()
+
+    def update(self, events=None):
+        """
+        Update the status of the menu using external events.
+
+        :param events: Pygame events as a list
+        :type events: list
+        :return: True if mainloop must be stopped
+        :rtype: bool
+        """
+        # NOTE: For usage, all Menu accesors must be "self._actual"
+        assert isinstance(events, list)
+        break_mainloop = not self._actual._mainloop_loop
+
+        # Update mouse
+        _pygame.mouse.set_visible(self._actual._mouse_visible)
+
+        # Surface needs an update
+        menu_surface_needs_update = False
+
+        # Process events, check title
+        if self._actual._menubar.update(events):
+            break_mainloop = True
```

Comment:
Updating the menubar should not break the mainloop.
I think that the flag `break_mainloop ` should be renamed in `updated` (as in widgets)

In fact the `update()` don't known if loop shall be broken or not. If user action has closed the menu, it will checked in the `mainloop()`

Reply:
But what would mean updated?